api/src/main/java/jakarta/data/repository/CrudRepository.java

```diff
@@ -107,9 +111,14 @@
     void deleteById(K id);
 
     /**
-     * Deletes a given entity.
+     * Deletes a given entity. Deletion is performed by matching the Id, and if the entity is
+     * versioned (for example, with {@code jakarta.persistence.Version}), then also the version.
```

Comment:
We should return to the discussion of creating a Repository specialization for JPA

Reply:
The JPA class is mentioned as an example.  If you have example classes from NoSQL or other patterns, I'll gladly include them, too.